File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/feedback/LikeEntity.java
package dev.ragnarok.fenrir.db.model.entity.feedback;

import com.google.gson.annotations.SerializedName;

import dev.ragnarok.fenrir.db.model.entity.Entity;
import dev.ragnarok.fenrir.db.model.entity.EntityWrapper;
import dev.ragnarok.fenrir.model.feedback.FeedbackType;

public class LikeEntity extends FeedbackEntity {

    @SerializedName("likesOwnerIds")
    private int[] likesOwnerIds;

    @SerializedName("liked")
    private EntityWrapper liked = new EntityWrapper();

    @SuppressWarnings("unused")
    public LikeEntity() {
    }

    public LikeEntity(@FeedbackType int type) {
        setType(type);
    }

    public int[] getLikesOwnerIds() {
        return likesOwnerIds;
    }

    public LikeEntity setLikesOwnerIds(int[] likesOwnerIds) {
        this.likesOwnerIds = likesOwnerIds;
        return this;
    }

    public Entity getLiked() {
        return liked.get();
    }

    public LikeEntity setLiked(Entity liked) {
        this.liked = new EntityWrapper().wrap(liked);
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/feedback/LikeEntity.kt
package dev.ragnarok.fenrir.db.model.entity.feedback

import androidx.annotation.Keep
import dev.ragnarok.fenrir.db.model.entity.DboEntity
import dev.ragnarok.fenrir.model.feedback.FeedbackType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Keep
@Serializable
@SerialName("like")
class LikeEntity : FeedbackEntity {
    var likesOwnerIds: LongArray? = null
        private set

    private var liked: DboEntity? = null

    @Suppress("UNUSED")
    constructor()
    constructor(@FeedbackType type: Int) {
        this.type = type
    }

    fun setLikesOwnerIds(likesOwnerIds: LongArray?): LikeEntity {
        this.likesOwnerIds = likesOwnerIds
        return this
    }

    fun getLiked(): DboEntity? {
        return liked
    }

    fun setLiked(liked: DboEntity?): LikeEntity {
        this.liked = liked
        return this
    }
}